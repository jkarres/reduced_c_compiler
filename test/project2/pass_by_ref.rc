int i;

function : void incr(int & j) {
  ++j;
}


function : void swap(int & a, int & b) {
  int temp = a;
  a = b;
  b = temp;
}


function : void oneref(int a, int & b) {
  a = 0;
  b = 0;
}

function : void main() {
  cout << i << endl;
  incr(i);
  cout << i << endl;


  int a = 11, b = 99;

  cout << a << " " << b << endl;
  swap(a, b);
  cout << a << " " << b << endl;


  oneref(a, b);
  cout << a << " " << b << endl;

}
