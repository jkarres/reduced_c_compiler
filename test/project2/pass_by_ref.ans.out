0
1
11 99
99 11
99 0
